include/llvm/IR/Type.h

```diff
@@ -71,7 +71,8 @@ class Type {
     StructTyID,      ///< 12: Structures
     ArrayTyID,       ///< 13: Arrays
     PointerTyID,     ///< 14: Pointers
-    VectorTyID       ///< 15: SIMD 'packed' format, or other vector type
+    VectorTyID,      ///< 15: SIMD 'packed' format, or other vector type
+    UnionTyID        ///< 16: Unions
```

Comment:
I don't understand why this is necessary. Clang has never needed an IR Union type before, so I think adding one is concerning. Additionally, if we do require this, it changes DXIL so we'll need to evaluate the impact of this on other hardware vendors.

Reply:
As mentioned this was unnecessary and we've since removed it.